Automattic/wp-calypso
# Checkout: Remove old isPaymentMethodEnabled

#### Changes proposed in this Pull Request

New checkout has a function called `isPaymentMethodEnabled()`. There is an old version in `lib/cart-values` which this PR removes in favor of the new version. The main difference is that new checkout has its own list of payment methods that is maintained there and https://github.com/Automattic/wp-calypso/pull/47260 removes the old list. After that, the two functions are identical and there's no need to have both. In addition, this removal will make it easier later to remove `lib/cart-values` entirely.

Depends on https://github.com/Automattic/wp-calypso/pull/47216
This is part of a chain of related PRs. The next one is https://github.com/Automattic/wp-calypso/pull/47262

#### Testing instructions

Despite the number of changes in this PR, most of them are for old checkout which is unused. There's only one place in-use that this PR will actually touch.

- Verify that the country-specific contact details form is displayed if purchasing a domain in either FR or UK.
